projects/schematics/src/migrations/2_0/constructor-deprecations/data/payment-form.component.migration.ts

```diff
@@ -0,0 +1,51 @@
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+import {
+  PAYMENT_FORM_COMPONENT,
+  SPARTACUS_STOREFRONTLIB,
+  USER_ADDRESS_SERVICE,
+  SPARTACUS_CORE,
+  CHECKOUT_PAYMENT_SERVICE,
+  CHECKOUT_DELIVERY_SERVICE,
+  USER_PAYMENT_SERVICE,
+  GLOBAL_MESSAGE_SERVICE,
+  FORM_BUILDER,
+  ANGULAR_FORMS,
+  MODAL_SERVICE,
+} from '../../../../shared/constants';
+
+export const PAYMENT_FORM_COMPONENT_MIGRATION: ConstructorDeprecation = {
+  class: PAYMENT_FORM_COMPONENT,
+  importPath: SPARTACUS_STOREFRONTLIB,
+  deprecatedParams: [
+    {
+      className: CHECKOUT_PAYMENT_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: CHECKOUT_DELIVERY_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: USER_PAYMENT_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: GLOBAL_MESSAGE_SERVICE,
+      importPath: SPARTACUS_CORE,
+    },
+    {
+      className: FORM_BUILDER,
+      importPath: ANGULAR_FORMS,
+    },
+    {
+      className: MODAL_SERVICE,
+      importPath: SPARTACUS_STOREFRONTLIB,
+    },
+  ],
+  addParams: [
+    {
+      className: USER_ADDRESS_SERVICE,
```

Comment:
To make this migration to work you should change order of parameters in contructor. UserAddressService should be the last parameter.

Other option is to change this migration. Add removeParams section with FormBuilder and ModalService, and then also add them to addParams after UserAddressService.

Reply:
And also mention this change in 2_0.md doc. Similarly to: PlaceOrderComponent

Reply:
Done and done